src/picture.rs

```diff
@@ -202,7 +215,10 @@ unsafe fn picture_alloc_with_edges(
         return res;
     }
     (*pic_ctx).allocator = (*p_allocator).clone();
-    (*pic_ctx).pic = (*p).clone();
+    // TODO(kkysen) A normal assignment here as it used to be
+    // calls `fn drop` on `(*pic_ctx).pic`, which segfaults as it is uninitialized.
+    // We need to figure out the right thing to do here.
+    addr_of_mut!((*pic_ctx).pic).write((*p).clone());
```

Comment:
What's the `addr_of_mut!` doing here?

Reply:
Getting a pointer to the field to do a write without creating a reference to uninitialized memory.